packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx

```diff
@@ -198,7 +199,8 @@ export function SavingsForm({ chainId, protocol, tab, onClose }: SavingsFormProp
                         })
                     }))
                 ) {
-                    throw new Error('Failed to deposit token.')
+                    // eslint-disable-next-line
+                    return Promise.reject(Promise.reject('Failed to deposit token.'))
```

Comment:
Why call `Promise.reject` twice?
```suggestion
                    return Promise.reject(new Error('Failed to deposit token.'))
```

Reply:
> 

This is a hack for AsyncFn reject.

Reply:
Can you explain what hack is it? I don't get it.